File: 网易2019实习生招聘编程题集合/数对.cpp
#include <bits/stdc++.h>

using namespace std;

typedef long long LL;

int main()
{
    for (LL n, k; cin >> n >> k; ) {
        LL ans = 0;
        if (!k)
            ans = n * n;
        else
            for (LL y = k + 1; y <= n; y++)
                ans += n / y * (y - k) + (n % y >= k ? n % y - k + 1 : 0);
        cout << ans << endl;
    }
    return 0;
}

File: 网易2019实习生招聘编程题集合/数对.java
import java.util.*;

public class Main {

	public static Scanner cin = new Scanner(System.in);

	public static void main(String[] args) {
		while (cin.hasNext()) {
			long n = cin.nextLong(), k = cin.nextLong(), ans = 0;
			if (k == 0)
				ans = n * n;
			else
				for (long y = k + 1; y <= n; y++)
					ans += n / y * (y - k) + (n % y >= k ? n % y - k + 1 : 0);
			System.out.println(ans);
		}
	}

}
